Doc/library/pathlib.rst

```diff
@@ -150,23 +150,49 @@ we also call *flavours*:
    to ``PurePosixPath('bar')``, which is wrong if ``foo`` is a symbolic link
    to another directory)
 
+   The optional *template* argument may provide another path object. It is
+   supplied whenever a new path object is created from an existing one, such
+   as in :attr:`parent` or :meth:`relative_to`. Subclasses may use this to
+   pass information between path objects. For example::
```

Comment:
Does it make sense to specify that ``template: Self | None`` here? I.e. that if template is not None, it will be an instance of the current (user-defined) class.

A